connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -190,6 +190,22 @@ class SparkSession(
     range(start, end, step, Option(numPartitions))
   }
 
+  def newSession(): SparkSession = {
+    throw new UnsupportedOperationException("newSession is not supported")
```

Comment:
Perhaps we should have a special exception for this, so it is easier to catch these compatibility issues for users. We can also use them to provide migration hints. Finally, there is also a difference between the methods we do want to support, and the ones we don't.

Reply:
This is to match Python client: https://github.com/apache/spark/blob/5243d0be2c15e3af36e981a9487ea600ab86a808/python/pyspark/sql/connect/session.py#L422


I see the difference between `want to support but not yet` and `won't support`. Right now it is not differentiated in both Scala client and Python client. We probably need a pass to change both together if we want to do so.

Reply:
For the record I think we should implement this particular method.

Regarding the errors in general. I don't think we spend a great deal of time on what we did on the python side. I am fine with coordinating efforts. Can you suggest something when you have cycles? Long term I want the never supported errors to point to something that offers some explanation as to why, and I want the will support someday errors to point to a JIRA. cc @grundprinzip

Reply:
I can follow up on this.